crates/bevy_time/src/lib.rs

```diff
@@ -41,6 +45,36 @@ impl Plugin for TimePlugin {
     }
 }
 
-fn time_system(mut time: ResMut<Time>) {
-    time.update();
+/// Channel resource used to receive time from render world
+pub struct TimeReceiver(pub Receiver<Instant>);
+/// Channel resource used to send time from render world
+pub struct TimeSender(pub Sender<Instant>);
+
+/// Creates channels used for sending time between render world and app world
+pub fn create_time_channels() -> (TimeSender, TimeReceiver) {
+    // bound the channel to 2 since when pipelined the render phase can finish before
+    // the time system runs.
+    let (s, r) = crossbeam_channel::bounded::<Instant>(2);
+    (TimeSender(s), TimeReceiver(r))
+}
+
+/// The system used to update the [`Time`] used by app logic. If there is a render world the time is sent from
+/// there to this system through channels. Otherwise the time is updated in this system.
+fn time_system(
+    mut time: ResMut<Time>,
+    time_recv: Option<Res<TimeReceiver>>,
+    mut has_received_time: Local<bool>,
+) {
+    if let Some(time_recv) = time_recv {
+        // TODO: delay checking channel on start by 2 frames when pipelined rendering
+        // is enabled. This is to make sure we always read the N-2 frame's time.
```

Comment:
I don't think I agree with the solution proposed in this comment. As such:
```suggestion
        // TODO: Figure out how to handle this when using pipelined rendering.
```